### DISCUSSION TITLE
Bivariate quantiles

### DISCUSSION THREAD

**Discussion by stephanvdw89** [kogalur/randomForestSRC#201] [Ideas]
I am using the rfsrc() and quantreg() functions to perform multivariate (bivariate) random forests. I am interested to calculate bivariate statistics from the quantiles (correlations, ratios), but only the marginal quantiles for each of the two output variables are provided in the output. Is there a way to get the bivariate quantiles (or cdf) from the quantreg outputs? See example code below for further clarification.

qmrf <- quantreg.rfsrc(fn,trn,mtry=M)
qmrf_pred <- quantreg.rfsrc(object=qmrf,newdata=tst)
tst_quants <- get.quantile(qmrf_pred)

tst_quants${Bivariate?}

**Comment by ishwaran** [comment]
Hi, it's an interesting question and idea.  Unfortunately the existing `quantreg` function doesn't have this capability.  The only way to get this would be to write your own function, but that would require you understanding what forest weights are and how to use those to obtain estimated values.  You would have to make a direct call to `rfsrc` using split rule `la.quantile.regr` with option `forest.wt=TRUE`.  That would be the first step.  Then you would have to use the `forest.wt` matrix  to calculate the bivariate statistics.
